src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -870,6 +976,43 @@ document::value delete_one(collection& coll, client_session* session, document::
     return result.extract();
 }
 
+document::value delete_many(collection& coll, client_session* session, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::view filter = arguments["filter"].get_document().value;
+    options::delete_options options{};
+
+    if (arguments["collation"]) {
+        options.collation(arguments["collation"].get_document().value);
+    }
+
+    if (arguments["hint"]) {
+        if (arguments["hint"].type() == bsoncxx::v_noabi::type::k_string)
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+
+    auto result = builder::basic::document{};
+    std::int32_t deleted_count = 0;
+
+    auto delete_result = [&] {
```

Comment:
Minor: is there a reason to use a lambda? Since `delete_result` is a `stdx::optional` I think it can be default initialized, so this this could follow the pattern of the `find_one_and_*` helpers:

```
stdx::optional<result::delete_result> delete_result;
if (session) {
	delete_result = ...
} else {
	delete_result = ...
}
```

Reply:
Sounds good to me; I just copied from `delete_one`. Technically one line shorter haha, but I do find the normal `if else` easier to read. Switched.